dbt/include/spark/macros/utils/split_part.sql

```diff
@@ -9,14 +9,34 @@
 
     {% endset %}
 
-    {% set split_part_expr %}
+    {% if part_number >= 0 %}
 
-    split(
-        {{ string_text }},
-        {{ delimiter_expr }}
-        )[({{ part_number - 1 }})]
+        {% set split_part_expr %}
 
-    {% endset %}
+        split(
+            {{ string_text }},
+            {{ delimiter_expr }}
+            )[({{ part_number - 1 if part_number > 0 else part_number }})]
+
+        {% endset %}
+
+    {% else %}
```

Comment:
Is there a test case that confirms that this works for non-negative numbers? If so, can we throw some negative numbers into that test case to show that it fails before the change and works after?

If that test case doesn't exist (it should exist, but I wouldn't be terribly surprised) could you create a test case for these scenarios?